File: leetcode/0014-longest-common-prefix/longestCommonPrefix.go
package leetcode

func longestCommonPrefix(strs []string) string {
	var prefix = ""
	if len(strs) < 1 {
		return prefix
	}

	var checkStr = strs[0]

	for index, letter := range checkStr {
		for strIndex, str := range strs {
			if strIndex == 0 {
				continue
			}

			if index >= len(str) || str[index] != byte(letter) {
				return prefix
			}
		}

		prefix = prefix + string(letter)
	}

	return prefix
}

File: leetcode/0014-longest-common-prefix/longestCommonPrefix.ts
function longestCommonPrefix(strs: string[]): string {
  let checkStr = strs[0] ?? "";
  let prefix = "";

  for (let index = 0; index < checkStr.length; index++) {
    const letter = checkStr[index];

    for (const str of strs) {
      if (str[index] !== letter) {
        return prefix;
      }
    }

    prefix = prefix + letter;
  }

  return prefix;
}

export default longestCommonPrefix;
